tslearn/shapelets.py

```diff
@@ -7,21 +7,41 @@
 from keras.models import Model
 from keras.layers import Dense, Conv1D, Layer, Input, concatenate, add
 from keras.metrics import categorical_accuracy, categorical_crossentropy, binary_accuracy, binary_crossentropy
+from keras.utils.generic_utils import get_custom_objects
 from sklearn.preprocessing import LabelBinarizer
-from sklearn.base import BaseEstimator, ClassifierMixin
+from sklearn.base import BaseEstimator, ClassifierMixin, TransformerMixin
+from sklearn.utils import check_array, column_or_1d
+from sklearn.utils.validation import check_is_fitted
+import keras
 from keras.regularizers import l2
 from keras.initializers import Initializer
 import keras.backend as K
 from keras.engine import InputSpec
 import numpy
 from tensorflow import set_random_seed
 
-from tslearn.utils import to_time_series_dataset
+import warnings
+
+from tslearn.utils import to_time_series_dataset, check_dims
 from tslearn.clustering import TimeSeriesKMeans
+from tslearn.preprocessing import TimeSeriesResampler
 
 __author__ = 'Romain Tavenard romain.tavenard[at]univ-rennes2.fr'
 
 
+# Patching unpickle_model to handle our custom layers
```

Comment:
If this piece of code is not needed, it can be removed

Reply:
Agreed